### ISSUE TITLE
`test.extend` triggers various rules from `all` config

### ISSUE THREAD

**Issue by gtbuchanan** [vitest-dev/eslint-plugin-vitest#686]
Calling `test.extend` and using the resulting test context results in unexpected warnings for the following rules:
* `require-top-level-describe`
* `require-hook`
* `no-standalone-expect`

```typescript
import { describe, test } from 'vitest';

interface Context { value: boolean }

const it = test.extend<Context>({ value: true }); // Triggers `require-top-level-describe`

describe.concurrent('extends', () => {
  it('should not trigger lint rules', ({ expect, value }) => { // Triggers `require-hook`
    expect(value).toBe(true); // Triggers `no-standalone-expect`
  });
});
```

Reproduction: [vitest-extend-eslint-bugs.zip](https://github.com/user-attachments/files/19475451/vitest-extend-eslint-bugs.zip)
